store/src/java/com/zimbra/cs/mailbox/MailItem.java

```diff
@@ -1914,7 +1919,11 @@ protected void finishCreation(MailItem parent) throws ServiceException {
         if (isLeafNode()) {
             boolean isDeleted = isTagged(Flag.FlagInfo.DELETED);
 
-            mMailbox.updateSize(mData.size, isQuotaCheckRequired());
+            if (isTrashFolder) {
+                mMailbox.updateSize(mData.size, isQuotaCheckRequired(), isTrashFolder);
+            } else {
+                mMailbox.updateSize(mData.size, isQuotaCheckRequired());
+            }
```

Comment:
It can be simplified without if-else.
`mMailbox.updateSize(mData.size, isQuotaCheckRequired(), isTrashFolder);`

note: please don't remove `void updateSize(long delta, boolean checkQuota) throws ServiceException` since it is not a private function.

Reply:
implemented as per suggestion, the updated method looks like below

```
    void updateSize(long delta, boolean checkQuota) throws ServiceException {
        updateSize(delta, checkQuota, false);
    }
```